apiserver/apiserver.go

```diff
@@ -596,7 +599,7 @@ func (srv *Server) endpoints() []apihttp.Endpoint {
 	)
 
 	add("/model/:modeluuid/applications/:application/resources/:resource", &ResourcesHandler{
-		StateAuthFunc: func(req *http.Request, tagKinds ...string) (ResourcesBackend, func(), names.Tag, error) {
+		StateAuthFunc: func(req *http.Request, tagKinds ...string) (ResourcesBackend, state.StatePoolReleaser, names.Tag, error) {
```

Comment:
This is from https://github.com/juju/juju/pull/7502/commits/5bc3e7a7981ffed96fc130986ee24e7b34a12943, but no functional change in this part of the code. The StatePoolReleaser has a boolean result that tells the caller whether or not releasing caused the State object to be closed and removed from the pool.